### ISSUE TITLE
memory_efficient_attention() non-deterministic behavior

### ISSUE THREAD

**Issue by takuma104** [facebookresearch/xformers#624]
# ❓ Questions and Help

Hi, 

First of all, thank you for creating a great library! 

I am using it in diffusers with stable-diffusion, but every time I use with xformers, it produces a picture where the generated result is very slightly different. I am investigating the root cause of this, but it seems that the memory_efficient_attention() results are not completely deterministic. The results are different for the same input in everytime. Is this a specification of memory_efficient_attention() or a bug? 

# environmnent

```
$ python -m xformers.info
A matching Triton is not available, some optimizations will not be enabled.
Error caught was: module 'triton.language' has no attribute 'constexpr'
xFormers 0.0.15.dev395+git.7e05e2c
memory_efficient_attention.cutlassF:               available
memory_efficient_attention.cutlassB:               available
memory_efficient_attention.flshattF:               available
memory_efficient_attention.flshattB:               available
memory_efficient_attention.smallkF:                available
memory_efficient_attention.smallkB:                available
memory_efficient_attention.tritonflashattF:        unavailable
memory_efficient_attention.tritonflashattB:        unavailable
swiglu.fused.p.cpp:                                available
is_triton_available:                               False
is_functorch_available:                            False
pytorch.version:                                   1.12.1
pytorch.cuda:                                      available
gpu.compute_capability:                            8.6
gpu.name:                                          NVIDIA GeForce RTX 3060
```

# reproduce code

```
import xformers
import xformers.ops
import torch

device = 'cuda'
dtype = torch.float16
shape = (1, 1024, 16, 16)

torch.manual_seed(0)
q = torch.rand(shape, device=device, dtype=dtype, requires_grad=False)
k = torch.rand(shape, device=device, dtype=dtype, requires_grad=False)
v = torch.rand(shape, device=device, dtype=dtype, requires_grad=False)

r1 = xformers.ops.memory_efficient_attention(q, k, v, attn_bias=None)
r2 = xformers.ops.memory_efficient_attention(q, k, v, attn_bias=None)

print(int((r1 != r2).sum())) # -> return non zero value
```

**Comment by takuma104** [comment]
I also performed a similar test on the original JAX implementation. In my environment (cuda, RTX3060) the results are all in perfect same in 10000 iterations.

https://gist.github.com/takuma104/29aa1eebd7e7fba611d8b1e872393114

**Comment by Tengxu-Sun** [comment]
I have got the same conclusion with you. Does it affect your final result?

**Comment by danthe3rd** [comment]
This is fixed in `0.0.19`, the kernels should now be deterministic.
Fixed in https://github.com/facebookresearch/xformers/commit/b63f634765414b03f5310adc140fa49eaeb2e6e7
